### PR TITLE
LifecycleNode::on_deactivate deactivate all managed entities.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/rclcpp/issues/1884

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

File: rclcpp_lifecycle/test/test_lifecycle_publisher.cpp [hunk 1]
```diff
@@ -80,7 +86,47 @@ class TestLifecyclePublisher : public ::testing::Test
   std::shared_ptr<EmptyLifecycleNode> node_;
 };
 
+TEST_F(TestLifecyclePublisher, publish_managed_by_node) {
```
Comment (fujitatomoya) [inline rclcpp_lifecycle/test/test_lifecycle_publisher.cpp:89]: verified this test case passes with this PR only.

**Comment by fujitatomoya** [comment]
@ivanpauno requesting review.

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16087/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10733/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16422/)

**Review by alsora — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@ivanpauno thanks for the follow-up verification, i will go ahead to merge this.

**Pull request merged**